### PR TITLE
Provide ScaleDownStatusProcessor with info about unremovable nodes

### PR DESCRIPTION
**Pull request by towca**

### CODE REVIEW

File: cluster-autoscaler/core/scale_down.go [hunk 1]
```diff
@@ -648,6 +666,18 @@ func (sd *ScaleDown) updateUnremovableNodes(nodes []*apiv1.Node) {
 	}
 }
 
+func (sd *ScaleDown) clearUnremovableNodeReasons() {
+	sd.unremovableNodeReasons = make(map[string]*simulator.UnremovableNode)
+}
+
+func (sd *ScaleDown) addUnremovableNodeReasonSimple(node *apiv1.Node, reason simulator.UnremovableReason) {
```
Comment (aleksandra-malinowska) [inline cluster-autoscaler/core/scale_down.go]: nit: consider renaming to 'addUnremovableNodeReason'

File: cluster-autoscaler/core/scale_down.go [hunk 2]
```diff
@@ -648,6 +666,18 @@ func (sd *ScaleDown) updateUnremovableNodes(nodes []*apiv1.Node) {
...
+	sd.unremovableNodeReasons[node.Name] = &simulator.UnremovableNode{Node: node, Reason: reason, BlockingPod: nil}
+}
+
+func (sd *ScaleDown) addUnremovableNodeReasonFull(unremovableNode *simulator.UnremovableNode) {
```
Comment (aleksandra-malinowska) [inline cluster-autoscaler/core/scale_down.go]: nit: consider renaming to 'addUnremovableNode'

File: cluster-autoscaler/core/scale_down.go [hunk 3]
```diff
@@ -779,45 +810,53 @@ func (sd *ScaleDown) TryToScaleDown(allNodes []*apiv1.Node, pods []*apiv1.Pod, p
...
 
 			nodeGroup, err := sd.context.CloudProvider.NodeGroupForNode(node)
 			if err != nil {
 				klog.Errorf("Error while checking node group for %s: %v", node.Name, err)
+				sd.addUnremovableNodeReasonSimple(node, simulator.UnexpectedError)
 				continue
 			}
 			if nodeGroup == nil || reflect.ValueOf(nodeGroup).IsNil() {
 				klog.V(4).Infof("Skipping %s - no node group config", node.Name)
+				sd.addUnremovableNodeReasonSimple(node, simulator.UnexpectedError)
```
Comment (aleksandra-malinowska) [inline cluster-autoscaler/core/scale_down.go]: This is expected to happen for all nodes from non-autoscaled groups, so I'm not really happy about mixing it up with actual errors. Let's either skip it or use a specific reason.

File: cluster-autoscaler/core/scale_down.go (hunk 1 above)
Comment (towca) [reply cluster-autoscaler/core/scale_down.go]: Done

File: cluster-autoscaler/core/scale_down.go (hunk 2 above)
Comment (towca) [reply cluster-autoscaler/core/scale_down.go]: Done

File: cluster-autoscaler/core/scale_down.go (hunk 3 above)
Comment (towca) [reply cluster-autoscaler/core/scale_down.go]: Changed to a specific reason.

File: cluster-autoscaler/simulator/drain.go [hunk 4]
```diff
@@ -35,8 +35,8 @@ import (
 // along with their pods (no abandoned pods with dangling created-by annotation). Useful for fast
 // checks.
 func FastGetPodsToMove(nodeInfo *schedulernodeinfo.NodeInfo, skipNodesWithSystemPods bool, skipNodesWithLocalStorage bool,
-	pdbs []*policyv1.PodDisruptionBudget) ([]*apiv1.Pod, error) {
-	pods, err := drain.GetPodsForDeletionOnNodeDrain(
+	pdbs []*policyv1.PodDisruptionBudget) ([]*apiv1.Pod, *drain.BlockingPod, error) {
```
Comment (aleksandra-malinowska) [inline cluster-autoscaler/simulator/drain.go]: Not in scope of this PR, but it seems like blocking pod and error could be combined (and error messages lazily evaluated - we discard them in some cases).

**Pull request merged**